src/util/fields.py

```diff
@@ -147,7 +147,12 @@ def clean_context(context):
         [model, r"\y{}\y".format(fieldname)],
     )
     for id_, name, context_s in cr.fetchall():
-        context = safe_eval(context_s or "{}", SelfPrintEvalContext(), nocopy=True)
+        try:
+            context = safe_eval(context_s or "{}", SelfPrintEvalContext(), nocopy=True)
+        except SyntaxError:
+            cr.execute("DELETE FROM ir_filters WHERE id = %s", [id_])    
```

Comment:
```suggestion
            cr.execute("DELETE FROM ir_filters WHERE id = %s", [id_])
```

Don't your editor automatically remove the trailing spaces?